Plugins/PluginsShared/PluginError.swift

```diff
@@ -0,0 +1,125 @@
+import PackagePlugin
+import Foundation
+
+enum PluginError: Swift.Error, CustomStringConvertible, LocalizedError {
+    case incompatibleTarget(targetName: String)
+    case noTargetOrDependenciesWithExpectedFiles(targetName: String, dependencyNames: [String])
+    case badArguments([String])
+    case noTargetsMatchingTargetName(targetName: String)
+    case tooManyTargetsMatchingTargetName(targetName: String, matchingTargetNames: [String])
+    case fileErrors([FileError])
+
+    var description: String {
+        switch self {
+        case .incompatibleTarget(let targetName):
+            return "Incompatible target called '\(targetName)'. Only Swift source targets can be used with the Swift OpenAPI Generator plugin."
+        case .noTargetOrDependenciesWithExpectedFiles(let targetName, let dependencyNames):
+            let introduction = dependencyNames.isEmpty ? "Target called '\(targetName)' doesn't contain" : "Target called '\(targetName)' or its local dependencies \(dependencyNames) don't contain"
+            return "\(introduction) any config or document files with expected names. For OpenAPI code generation, a target needs to contain a config file named 'openapi-generator-config.yaml' or 'openapi-generator-config.yml', as well as an OpenAPI document named 'openapi.yaml', 'openapi.yml' or 'openapi.json' under target's source directory. See documentation for details."
```

Comment:
```suggestion
            return "\(introduction) any config or OpenAPI document files with expected names. See documentation for details."
```

Let's add "OpenAPI" to the sentence and drop the summary of the docs, we don't want to have to keep it in sync here with the other places. Sending people to docs is good enough.